lib/propcheck/output_agent.ex

```diff
@@ -0,0 +1,36 @@
+defmodule PropCheck.OutputAgent do
+  @moduledoc """
+  An agent to gather unique PropCheck-internal output from tests.
+  """
+  use Agent
+
+  @doc """
+  Start the agent.
+  """
+  def start_link do
+    Agent.start_link(&MapSet.new/0)
```

Comment:
I use a `MapSet` here to reduce the verbosity on shrinking. Otherwise, if we hit the exception on every attempt to shrink, the output will be large. Note however that the stacktraces in the `assert_test.exs` are apparently not always the same, so there is still more output than there should be.